### PR TITLE
AI Tutor: convert tutor variations components to TypeScript

### PR DESCRIPTION
**Pull request by Erin007**
[CT-301](https://codedotorg.atlassian.net/browse/CT-301)

Code Tools is working on converting our components (at least the new ones) to TypeScript based on this [proposal](https://docs.google.com/document/d/15KLtvSTHFOGIVwu60l1mFZscFL8miXvZHKNCKwuzf4E/edit#heading=h.a1916db8qd8j).  To support this initiative, as a mini-project for Meetings Week, the AI Tutor team is working on converting the files in `apps/src/code-studio/components/aiTutor` to TypeScript.  This is continuation of the work in #56103.

This PR tackles the components related to specific tutor variations: `generalChatTutor`, `validationTutor` and `compilationTutor`, as well as `warningModal`. 

I've confirmed that the tutor variations still work as expected with these changes. 

Follow up work will include converting the remaining components. #55937 deletes `aiTutorPanelContainer`, creates  `aiTutorFloatingActionButton` and modifies the `aiTutorPanel` component; those will be converted to TypeScript after #55937 is merged to avoid conflicts/confusion. [CT-302](https://codedotorg.atlassian.net/browse/CT-302)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by molly-moen — APPROVED** [review]
Nice!

File: apps/src/code-studio/components/aiTutor/compilationTutor.tsx
Comment (sanchitmalhotra126) [inline apps/src/code-studio/components/aiTutor/compilationTutor.tsx:18]: Something I've been using recently is `TypedUseSelectorHook`, which allows you to create a `useSelector` with the typings on the state object that you need, for example
```
const useTypedSelector: TypedUseSelectorHook<{javalabEditor: JavalabEditorState; aiTutor: AITutorState}> = useSelector;
...

// knows what type state is automatically
const sources = useTypedSelector(state => state.javalabEditor.sources);
const aiResponse = useTypedSelector(state => state.aiTutor.aiResponse);
```
It can be overkill sometimes but I find it helpful when you have a lot of `useSelectors`. Not a requirement, but just thought i'd share if you find it useful!

(Also thinking out loud a bit, if folks do find it useful, we could even vend out typed useSelector hooks with each redux store, like `useJavalabEditorSelector`, `useAiTutorSelector`, etc, that are automatically typed...though I suppose this would only be useful until we have a global store type. @molly-moen curious what you think though)

**Review by sanchitmalhotra126 — APPROVED** [review]
One optional suggestion but LGTM!

File: apps/src/code-studio/components/aiTutor/compilationTutor.tsx
Comment (molly-moen) [reply apps/src/code-studio/components/aiTutor/compilationTutor.tsx:18]: So after reviewing this PR I thought to myself "why don't we just make a global type for our redux store" and started doing that 😆 . I should have a PR pretty shortly with the creation of that type, after seeing this I'll also add a `TypedUseSelectorHook` that we can use too!

Comment (molly-moen) [reply apps/src/code-studio/components/aiTutor/compilationTutor.tsx:18]: https://github.com/code-dot-org/code-dot-org/pull/56208

Comment (molly-moen) [reply apps/src/code-studio/components/aiTutor/compilationTutor.tsx:18]: (not blocking on this pr, we can follow up and clean this up if desired)

Comment (Erin007) [reply apps/src/code-studio/components/aiTutor/compilationTutor.tsx:18]: Ooooh. I like the look of this. Seems much more streamlined. I'm going to skip incorporating on this PR because I'm in a PR domino situation and want to unblock myself from merge conflicts on other PRs 🙃, but made a note to update on my "Future AI Tutor work items" list. Thank you, both!

**Pull request merged**
